packages/promotion/src/repositories/campaign.ts

```diff
@@ -8,4 +10,137 @@ export class CampaignRepository extends DALUtils.mikroOrmBaseRepositoryFactory<
     create: CreateCampaignDTO
     update: UpdateCampaignDTO
   }
->(Campaign) {}
+>(Campaign) {
+  async create(
+    data: CreateCampaignDTO[],
+    context: Context = {}
+  ): Promise<Campaign[]> {
+    const manager = this.getActiveManager<SqlEntityManager>(context)
+    const promotionIds: string[] = []
+    const campaignIdentifierPromotionsMap = new Map<string, string[]>()
+
+    data.forEach((campaignData) => {
+      const campaignPromotionIds =
+        campaignData.promotions?.map((p) => p.id) || []
+
+      promotionIds.push(...campaignPromotionIds)
+
+      campaignIdentifierPromotionsMap.set(
+        campaignData.campaign_identifier,
+        promotionIds
+      )
+
+      delete campaignData.promotions
+    })
+
+    const existingPromotions = await manager.find(Promotion, {
+      id: promotionIds,
+    })
+
+    const existingPromotionsMap = new Map<string, Promotion>(
+      existingPromotions.map((promotion) => [promotion.id, promotion])
+    )
+
+    const createdCampaigns = await super.create(data, context)
+
+    for (const createdCampaign of createdCampaigns) {
+      const campaignPromotionIds =
+        campaignIdentifierPromotionsMap.get(
+          createdCampaign.campaign_identifier
+        ) || []
+
+      if (!campaignPromotionIds.length) {
+        continue
+      }
```

Comment:
nit: this is redundant since `campaignPromotionIds` has no items and the for-loop gets "skipped". Feel free to keep it for clarity if you fancy 😄